--- Vrrpd-master/Makefile
+++ vrrpd-master/Makefile
@@ -18,12 +18,13 @@
 
 # set to compile with GCC
 CC=gcc
-CFLAGS= $(COMMON_CFLAGS) -Wall 
+CFLAGS+= $(COMMON_CFLAGS) -Wall 
+LDFLAGS+= $(LINKLIB)
 
 all: $(ALL_EXE)
 
 vrrpd : $(VRRPD_OBJS)
-	$(CC) $(PROF_OPT) -o $@ $^ -lrt $(LINKLIB)
+	$(CC) $(PROF_OPT) -o $@ $^ -lrt $(LDFLAGS)
 
 atropos:$(ATROPOS_OBJS)
 	$(CC) $(PROF_OPT) -o $@ $^ -lrt

--- Vrrpd-master/vrrpd.c
+++ vrrpd-master/vrrpd.c
@@ -1535,6 +1540,7 @@
 	int		len	= vrrp_read( vsrv, buf, sizeof(buf) );
 	struct iphdr	*iph	= (struct iphdr *)buf;
 	vrrp_pkt	*hd	= (vrrp_pkt *)((char *)iph + (iph->ihl<<2));
+	int delay	= 0;
 
 	if( vsrv->state != VRRP_STATE_BACK ){
 		if( vsrv->wantstate == VRRP_STATE_BACK ){
@@ -1554,7 +1560,7 @@
 		}else if( hd->priority > vsrv->priority ||
 				(hd->priority == vsrv->priority &&
 				ntohl(iph->saddr) > vsrv->vif.ipaddr) ){
-			int delay	= 3*vsrv->adver_int + VRRP_TIMER_SKEW(vsrv);
+			delay	= 3*vsrv->adver_int + VRRP_TIMER_SKEW(vsrv);
 	be_backup:
 			state_leave_master( vsrv, 0 );
 			vsrv->state	= VRRP_STATE_BACK;
