--- Vrrpd-master/vrrpd.c
+++ vrrpd-master/vrrpd.c
@@ -50,6 +50,8 @@
 #include "ipaddr.h"
 
 #ifdef DLINK_SOFTWARE
+#include <jansson.h>
+#include <stdbool.h>
 #include <d-link/d_service_notify.h>
 #endif
 
@@ -1393,6 +1395,10 @@
 	
 	/* add the ip addresses */
 	ipaddr_ops( vsrv, 1 );
+
+#ifdef DLINK_SOFTWARE
+	dlink_service_update(json_pack("{s:b}", "Master", true));
+#endif
 	
 	/* send an advertisement */
 	vrrp_send_adv( vsrv, vsrv->priority );
@@ -1447,6 +1453,10 @@
 	}
 	/* remove the ip addresses */
 	ipaddr_ops( vsrv, 0 );
+
+#ifdef DLINK_SOFTWARE
+	dlink_service_update(json_pack("{s:b}", "Master", false));
+#endif
 
 	/* if we stop vrrpd, warn the other routers to speed up the recovery */
 	if( advF ){

