diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 07be5d9..1742ae5 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -82,5 +82,5 @@ install(TARGETS libmosquitto RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION
 install(FILES mosquitto.h DESTINATION "${INCLUDEDIR}")
 
 if (UNIX)
-	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
+#	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
 endif (UNIX)
diff --git a/lib/cpp/CMakeLists.txt b/lib/cpp/CMakeLists.txt
index b80dc18..491e121 100644
--- a/lib/cpp/CMakeLists.txt
+++ b/lib/cpp/CMakeLists.txt
@@ -14,5 +14,5 @@ install(TARGETS mosquittopp RUNTIME DESTINATION "${BINDIR}" LIBRARY DESTINATION
 install(FILES mosquittopp.h DESTINATION "${INCLUDEDIR}")
 
 if (UNIX)
-	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
+#	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
 endif (UNIX)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6918ac2..8f1a19e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -141,6 +141,6 @@ if (${WITH_TLS} STREQUAL ON)
 endif (${WITH_TLS} STREQUAL ON)
 
 if (UNIX)
-	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
+#	install(CODE "EXEC_PROGRAM(/sbin/ldconfig)")
 endif (UNIX)
 
