From 459fa3fd129ac6c47fcc5571176864fb33dcf9b7 Mon Sep 17 00:00:00 2001
From: Maxim Zhukov <mzhukov@dlink.ru>
Date: Mon, 21 Jan 2019 18:07:35 +0300
Subject: [#33347] Makefile: do not build lua as static
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Т.к. у нас уже есть lua в системе, нет смысла использовать Lua из репа,
т.к. это добавляет ~200к

Signed-off-by: Maxim Zhukov <mzhukov@dlink.ru>
---
 Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index acbafc8..6beadcd 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,6 @@
-LUA     = lua-5.1.4
 STATIC  = false
 
-FLAGS   = -fno-exceptions -fno-rtti -O2 -I$(LUA) -L$(LUA) $(CFLAGS)
+FLAGS   = -fno-exceptions -fno-rtti -O2 $(CFLAGS)
 SRC     = main.cpp soap.cpp mem.cpp mcast.cpp luaxlib.cpp luaxcore.cpp luajson.cpp luajson_parser.cpp
 LUAMYCFLAGS = -DLUA_USE_LINUX
 
@@ -11,7 +10,6 @@ LUAMYCFLAGS+=-static
 endif
 
 all:
-	make -C $(LUA) a
 	$(CC) -O2 -c -o md5.o md5c.c
 	$(CC) $(FLAGS) -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -o xupnpd $(SRC) md5.o -llua -ldl -lm
 	# -DWITH_LIBUUID  -luuid 
@@ -32,8 +30,6 @@ all:
 	ln -snf  $/../../../tmp/xupnpd.lua $(TARGET_DIR)/usr/share/xupnpd/xupnpd.lua
 
 clean:
-	make -C $(LUA) clean
-	rm -f $(LUA)/liblua.a
 	rm -f md5.o
 	rm -f xupnpd
 	rm -f $(TARGET_DIR)/usr/share/xupnpd/xupnpd.lua
-- 
2.20.1

