diff --git a/src/dhcpv6.c b/src/dhcpv6.c
index 9dce577..e7020ea 100644
--- a/src/dhcpv6.c
+++ b/src/dhcpv6.c
@@ -613,7 +613,7 @@ int dhcpv6_request(enum dhcpv6_msg type)
 		// Receive rounds
 		for (; len < 0 && (round_start < round_end);
 				round_start = odhcp6c_get_milli_time()) {
-			uint8_t buf[1536];
+			uint8_t buf[4608];
 			union {
 				struct cmsghdr hdr;
 				uint8_t buf[CMSG_SPACE(sizeof(struct in6_pktinfo))];
