--- a/src/main.c
+++ b/src/main.c
@@ -48,6 +48,9 @@
 #include "iperf_locale.h"
 #include "net.h"
 
+#ifdef DLINK_SOFTWARE
+#include <d-link/d_service_notify.h>
+#endif
 
 static int run(struct iperf_test *test);
 
@@ -104,6 +107,10 @@
         exit(1);
     }
 
+#ifdef DLINK_SOFTWARE
+    dlink_service_connect(0);
+#endif
+
     if (run(test) < 0)
         iperf_errexit(test, "error - %s", iperf_strerror(i_errno));
 
